src/operations/charts/LineChartOperations.ts

```diff
@@ -0,0 +1,227 @@
+ * and then generate the chart options for Highcharts.
+ */
+export default class LineChartOperations {
+  /**
+   * The first four line colors are fixed; if more than four lines are rendered,
+   * the default Highcharts colors will be used.
+   */
+  private static LINE_COLORS = ['#FFB74D', '#157DBC', '#85E77C', '#FF5252'];
+
+  /**
+   * With this static function, the LineChart component can ensure that the received data for the chart
+   * is converted to the `LineChartData` type.
+   * To support another interface in `LineChartProps.data`, one has to add another switch case here
+   * that converts the new interface into `LineChartData`.
+   * @param data
+   */
+  public static convertToLineChartData(
+    data: LineChartData | BalanceOfTradeGraph | CurrencyExchangeGraph | InflationGraphs
+  ): LineChartData {
+    switch (data.type) {
+      case 'LineChartData':
+        return data;
+
+      case 'BalanceOfTradeGraph':
+        return {
+          type: 'LineChartData',
+          xAxisType: 'datetime',
+          lines: [
+            {
+              name: 'Balance of Trade',
+              dataPoints: data.data.map((p) => {
+                return { x: p.x, y: p.y };
+              }),
+            },
+          ],
+        };
+
+      case 'CurrencyExchangeGraph':
+        return {
+          type: 'LineChartData',
+          xAxisType: 'datetime',
+          lines: [
+            {
+              name: data.name,
+              dataPoints: data.data.map((p) => {
+                return { x: p.x, y: p.y };
+              }),
+            },
+          ],
+        };
+
+      case 'InflationGraphs':
+        return {
+          type: 'LineChartData',
+          xAxisType: 'datetime',
+          lines: [
+            {
+              name: 'Headline Inflation',
+              dataPoints: data.headline.data.map((p) => {
+                return { x: p.x, y: p.y };
+              }),
+            },
+            {
+              name: 'Food Inflation',
+              dataPoints: data.food.data.map((p) => {
+                return { x: p.x, y: p.y };
+              }),
+            },
+          ],
+        };
+
+      default:
+        // if the given type is not supported yet, an empty `LineChartData` instance is returned
+        return {
+          type: 'LineChartData',
+          xAxisType: 'linear',
+          lines: [],
+        };
+    }
+  }
+
+  /**
+   * With this static function, the LineChart component can build the HighCharts options,
+   * needed for the HighCharts component, out of a given `LineChartData` instance.
+   * It is expected that all line data in `LineChartData.lines` have the same `x` values and provide
+   * a `y` value for each `x` value. For example, if one line has values for x=1, x=2, and x=3,
+   * the second line must also provide `y` values for these exact `x` values and no more or less.
+   */
+  public static getHighChartData(data: LineChartData): Highcharts.Options {
```

Comment:
pass the theme via the parameters, you should avoid using hooks in an operations file